nvflare/app_opt/xgboost/tree_based/shareable_generator.py

```diff
@@ -24,20 +24,37 @@
 
 def update_model(prev_model, model_update):
     if not prev_model:
-        model_update["learner"]["gradient_booster"]["model"]["gbtree_model_param"]["num_parallel_tree"] = "1"
         return model_update
     else:
-        # Always 1 tree, so [0]
-        best_iteration = int(prev_model["learner"]["attributes"]["best_iteration"])
-        best_ntree_limit = int(prev_model["learner"]["attributes"]["best_ntree_limit"])
-        num_trees = int(prev_model["learner"]["gradient_booster"]["model"]["gbtree_model_param"]["num_trees"])
-        prev_model["learner"]["attributes"]["best_iteration"] = str(best_iteration + 1)
-        prev_model["learner"]["attributes"]["best_ntree_limit"] = str(best_ntree_limit + 1)
-        prev_model["learner"]["gradient_booster"]["model"]["gbtree_model_param"]["num_trees"] = str(num_trees + 1)
-        append_info = model_update["learner"]["gradient_booster"]["model"]["trees"][0]
-        append_info["id"] = num_trees
-        prev_model["learner"]["gradient_booster"]["model"]["trees"].append(append_info)
-        prev_model["learner"]["gradient_booster"]["model"]["tree_info"].append(0)
+        # Append all trees
+        pre_num_parallel_tree = int(
+            prev_model["learner"]["gradient_booster"]["model"]["gbtree_model_param"]["num_parallel_tree"]
+        )
+        pre_best_iteration = int(prev_model["learner"]["attributes"]["best_iteration"])
+        pre_best_ntree_limit = int(prev_model["learner"]["attributes"]["best_ntree_limit"])
+        pre_num_trees = int(prev_model["learner"]["gradient_booster"]["model"]["gbtree_model_param"]["num_trees"])
+
+        add_num_parallel_tree = int(
+            model_update["learner"]["gradient_booster"]["model"]["gbtree_model_param"]["num_parallel_tree"]
+        )
+        add_best_iteration = int(model_update["learner"]["attributes"]["best_iteration"])
+        add_best_ntree_limit = int(model_update["learner"]["attributes"]["best_ntree_limit"])
+        add_num_trees = int(model_update["learner"]["gradient_booster"]["model"]["gbtree_model_param"]["num_trees"])
```

Comment:
can you create a method to wrap this?
for example:

```
def _get_xgboost_model_attr(xgb_model):
    num_parallel_tree = int(XXX)
    best_iteration = int(XXX)
    best_ntree_limit = int(XXX)
    num_trees = int(XXX)
    return num_parallel_tree, best_iteration, best_ntree_limit, num_trees 
```

and just call this method with prev_model or model_update

Reply:
good idea, added